src/sentry/profiles/flamegraph.py

```diff
@@ -559,3 +561,96 @@ def _query_chunks_for_profilers(self, query: Query) -> Mapping[str, Any]:
         )
 
         return raw_snql_query(request, referrer=referrer)
+
+    def get_profile_candidates_from_profiles(self) -> ProfileCandidates:
+        if self.snuba_params.organization is None:
+            raise ValueError("`organization` is required and cannot be `None`")
+
+        max_profiles = options.get("profiling.flamegraph.profile-set.size")
+
+        referrer = Referrer.API_PROFILING_PROFILE_FLAMEGRAPH_PROFILE_CANDIDATES.value
+
+        transaction_profiles_builder = DiscoverQueryBuilder(
+            dataset=Dataset.Discover,
```

Comment:
🤔 do you get a bit more perf by doing `Dataset.Transactions` or are there profiles on Error events too?

Reply:
No profiles on error events. Not right now anyways. So I guess I could switch it and it'd be the same 🤔 I'm going to merge this as is and follow up to see how that works.

Reply:
👍 I *think* its slightly faster, cause snuba doesn't need to make any dataset decision, and then the transactions table has less rows